--- output/face-gpu.Dockerfile	2023-09-01 10:05:15.315939418 +0200
+++ face-gpu.Dockerfile_fix	2023-09-01 10:24:36.368086964 +0200
@@ -10,6 +10,11 @@
 ENV MXNET_CUDNN_AUTOTUNE_DEFAULT=0
 ENV MXNET_CUDNN_LIB_CHECKING=0
 
+# fix key issues, copied from https://github.com/NVIDIA/nvidia-docker/issues/1632#issuecomment-1125739652
+RUN apt-key del 7fa2af80
+RUN apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/3bf863cc.pub
+RUN apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu2004/x86_64/7fa2af80.pub
+
 RUN apt-get -y update && apt-get install -y \
     software-properties-common \
     build-essential \
@@ -33,7 +38,7 @@
     python3-pip \
     python3-setuptools \
     unzip
-
+    
 # some image/media dependencies
 RUN apt-get -y update && apt-get install -y \
     libjpeg8-dev \
